humility-core/src/hubris.rs

```diff
@@ -3034,6 +3034,38 @@ impl HubrisArchive {
         !self.modules.is_empty()
     }
 
+    ///
+    /// Takes a list of potentially similar types and deduplicates the list.
+    ///
+    fn dedup(&self, goffs: Vec<HubrisGoff>) -> Result<Vec<HubrisGoff>> {
+        let mut cmp = Vec::new();
+        let mut out = Vec::new();
+
+        if goffs.len() <= 1 {
```

Comment:
Nit: this conditional isn't strictly necessary, since a 1-item `Vec` would dedup to itself.

Reply:
Yeah, that was a (lame) attempt to avoid the unnecessary `lookup_type` in the (common) case of a single element.